proteus/mprans/BoundaryConditions.pyx

```diff
@@ -319,64 +319,70 @@ class BC_RANS(BC_Base):
-    #         if x[vert_axis] < waterLevel:
-    #             return u_p
-    #         elif x[vert_axis] >= waterLevel:
-    #             return None
-
-    #     self.u_dirichlet = get_inlet_ux_dirichlet(U[0])
-    #     self.v_dirichlet = get_inlet_ux_dirichlet(U[1])
-    #     if len(U) == 3:
-    #             self.w_dirichlet = get_inlet_ux_dirichlet(U[2])
-    #     self.vof_dirichlet = inlet_vof_dirichlet
-    #     self.p_advective = inlet_p_advective
+    # FOLLOWING BOUNDARY CONDITION IS UNTESTED #
+    def setTwoPhaseVelocityInlet(self, U, waterLevel, vert_axis=None, air=1.,
+                                 water=0.):
+        """
+        Imposes a velocity profile lower than the sea level and an open
+        boundary for higher than the sealevel.
+        :param U: Velocity vector at the global system.
+        :param waterLevel: water level at global coordinate system.
+        :param vert_axis: index of vertical in position vector, must always be
+                        aligned with gravity, by default set to 1].
+        :param air: Volume fraction for air (1.0 by default).
+        :param water: Volume fraction for water (0.0 by default).
+        Below the seawater level, the condition returns the _dirichlet and
+        p_advective condition according to the inflow velocity.
+        Above the sea water level, the condition returns the gravity as zero,
+        and sets _dirichlet condition to zero, only if there is a zero inflow
+        velocity component.
+        (!) This condition is best used for boundaries and gravity aligned with
+            one of the main axes.
+        """
+        self.reset()
+        self.BC_type = 'TwoPhaseVelocityInlet'
+
+        U = np.array(U)
+        if vert_axis is None:
+            vert_axis = self.Shape.Domain.nd - 1
+
+        def get_inlet_ux_dirichlet_cython(i):
+            def get_inlet_ux_dirichlet():
+                def ux_dirichlet(x, t):
+                    if x[vert_axis] < waterLevel:
+                        return U[i]
+                    elif x[vert_axis] >= waterLevel and U[i] == 0:
+                        return 0.
+                return ux_dirichlet
+            return get_inlet_ux_dirichlet
+
+        def inlet_vof_dirichlet_cython():
+            def inlet_vof_dirichlet(x, t):
+                if x[vert_axis] < waterLevel:
+                    return water
+                elif x[vert_axis] >= waterLevel:
+                    return air
+            return inlet_vof_dirichlet
+
+        def inlet_p_advective_cython():
+            def inlet_p_advective(x, t):
+                b_or = self._b_or[self._b_i]
+                u_p = np.sum(U * b_or)
+                # This is the normal velocity, based on the inwards boundary
+                # orientation -b_or
+                u_p = -u_p
+                if x[vert_axis] < waterLevel:
+                    return u_p
+                elif x[vert_axis] >= waterLevel:
+                    return None
```

Comment:
@nerisaurus, this is possible, as this boundary condition is untested, it looks like it should return a `[0,0,0]` array instead of `None` like for `setUnsteadyTwoPhaseVelocityInlet`